src/main/scala/firrtl/passes/memlib/ResolveMaskGranularity.scala

```diff
@@ -65,7 +65,8 @@ object AnalysisUtils {
       if (nodeWidth == extractionWidth) getOrigin(connects)(args.head) else e
     case DoPrim((PrimOps.AsUInt | PrimOps.AsSInt | PrimOps.AsClock), args, _, _) => 
       getOrigin(connects)(args.head)
-    case ValidIf(cond, value, ClockType) => getOrigin(connects)(value)
+    // It is a correct optimization to treat ValidIf as a connection
+    case ValidIf(cond, value, _) => getOrigin(connects)(value)
```

Comment:
Why the change of requirement for ClockType?

Reply:
Good question, so I traced the origin of this line to https://github.com/freechipsproject/firrtl/pull/310 where it was added to ensure we can optimize past ValidIfs on clocks. In doing this PR, I realized we need to optimize past ValidIfs for enable signals as well which are Bools. It's just a minor generalization.